stage4.1_first_application/pycasa/ui/tasks/pycasa_browser_pane.py

```diff
@@ -0,0 +1,27 @@
+# General imports
+
+# ETS imports
+from traits.api import HasStrictTraits, Instance, observe, Str
+from traitsui.api import InstanceEditor, Item, ModelView, View
+from pyface.tasks.api import EditorAreaPane, IEditor, IEditorAreaPane, \
+    PaneItem, Task, TraitsDockPane
+
+# Local imports
+from ..file_browser_view import FileBrowserView
+
+
+class PycasaBrowserPane(TraitsDockPane):
```

Comment:
Thoughts on implementing this in a single dock pane, l[ike this pyface example](https://github.com/enthought/pyface/blob/6c9ae8328acef522bffd71e1d26aa502de926b67/examples/tasks/basic/example_panes.py#L10-L39)?

```python
class FileBrowserPane(TraitsDockPane):
    """ A simple file browser pane.
    """


    # TaskPane interface ---------------------------------------------------


    id = "example.file_browser_pane"
    name = "File Browser"


    # FileBrowserPane interface --------------------------------------------


    # Fired when a file is double-clicked.
    activated = Event()


    # The list of wildcard filters for filenames.
    filters = List(Str)


    # The currently selected file.
    selected_file = File(os.path.expanduser("~"))


    # The view used to construct the dock pane's widget.
    view = View(
        Item(
            "selected_file",
            editor=FileEditor(dclick_name="activated", filter_name="filters"),
            style="custom",
            show_label=False,
        ),
        resizable=True,
    )
```

This way, there's less code that attendees will have to write and fit in their head, hopefully saving us time!

Reply:
Yes, I thought about it but decided to structure it the model, the view, the pane and the task in separate modules so classes are split and can scale better for a real application. It's definitely more of a philosophical decision. What do you think?

Reply:
Going to mark this as resolved for now and we can discuss this further.